compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/NodeCloner.java

```diff
@@ -991,7 +991,7 @@ public void visit(BLangLetExpression source) {
         BLangLetExpression clone = new BLangLetExpression();
         source.cloneRef = clone;
         clone.letVarDeclarations = cloneLetVarDeclarations(source.letVarDeclarations);
```

Comment:
Can't you use the `cloneList()` method for this?

Reply:
BLangLetVariable does not extends `Node`

`private <T extends Node> List<T> cloneList(List<T> nodes) {`